package.json

```diff
@@ -5,47 +5,41 @@
   "license": "MIT",
   "author": "ReadMe <support@readme.io> (https://readme.com)",
   "sideEffects": false,
+  "type": "module",
```

Comment:
Now that this field is here, `tsup` automatically detects that and flips the extensions so now:

`.js` / `.d.ts` = ESM
`.cjs` / `.d.cts` = CJS

Reply:
Correct me if im wrong but now the package is a full esm package

The correct way to fix it would be 


```
       "./webpack": {
            "require": {
                "types": "./dist/webpack.d.ts",
                "default": "./dist/webpack.js"
            },
            "import": {
                "types": "./dist/webpack.d.mts",
                "default": "./dist/webpack.mjs"
            }
        },
```

Reply:
One more example, https://github.com/storybookjs/storybook/pull/19269#discussion_r991455547

Reply:
And this tool can be used to check it https://github.com/arethetypeswrong/arethetypeswrong.github.io

Reply:
@prisis that tool is fantastic, thanks for sharing!

> The correct way to fix it

We don't need to explicitly define the type declaration files since TS will automatically pick up on the co-located files ([see here for more info](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing:~:text=If%20it%20finds%20them%2C%20it%20will%20look%20for%20a%20co%2Dlocated%20declaration%20file.)). This PR was published to `oas@21.1` and it appears that the types are being picked up by that website properly: https://arethetypeswrong.github.io/?p=oas@21.1

> Correct me if im wrong but now the package is a full esm package

Yeah if you read the PR description, we unfortunately needed to convert it over to ESM in order to get it working properly in another `module: node16` + ESM repo I'm working on. If you look at the "before" results in the image below, it wasn't working properly in ESM nor in CJS.

The good thing is that the website says that we're still shipping our main CJS exports with the same warnings as we were before (see the "`node16` from CJS" row on [this page](https://arethetypeswrong.github.io/?p=oas@21.1)). Based on the warnings warnings, my guess is that these are issues with our `tsup` configuration or the code itself and not our `package.json` `exports` definition.

**Before the `exports` fixes in this PR** ([`oas@21.0.3`](https://arethetypeswrong.github.io/?p=oas@21.0.3)):

![CleanShot 2023-09-13 at 10 29 25@2x](https://github.com/readmeio/oas/assets/8854718/016c5261-f71a-41cc-8586-f9b6fd0716ce)

**After the `exports` fixes in this PR** ([`oas@21.1.1`](https://arethetypeswrong.github.io/?p=oas@21.1.1)):

![CleanShot 2023-09-13 at 10 29 39@2x](https://github.com/readmeio/oas/assets/8854718/a4ff8f80-6a92-4abd-956e-7f729a37972f)